core/ui_events.js

```diff
@@ -54,22 +56,43 @@ Blockly.utils.object.inherits(Blockly.Events.Ui, Blockly.Events.Abstract);
  */
 Blockly.Events.Ui.prototype.IS_UI_EVENT = true;
 
+/**
+ * Class for a click event.
+ * @param {?Blockly.Block=} opt_block The affected block. Null for click events
+ *     that do not have an associated block (i.e. workspace click). Undefined
+ *     for a blank event.
+ * @param {string=} opt_workspaceId The workspace identifier for this event.
+ * @param {string=} opt_element One of 'block' or 'workspace'
+ * @extends {Blockly.Events.Ui}
+ * @constructor
+ */
+Blockly.Events.Click = function(opt_block, opt_workspaceId, opt_element) {
+  var workspaceId = opt_block ? opt_block.workspace.id : opt_workspaceId;
+  Blockly.Events.Click.superClass_.constructor.call(this, workspaceId);
+  this.blockId = opt_block ? opt_block.id : null;
+
+  /**
+   * The category of element targeted by this click event.
+   * @type {string}
+   */
+  this.element = opt_element ? opt_element :
+      this.isBlank ? '' : opt_block ? 'block' : 'workspace';
```

Comment:
I think I would prefer this in an if statement? The nested ternary is a bit difficult to read.

Reply:
Done.